haystack_experimental/components/routers/document_length_router.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, List
+
+from haystack import component
+from haystack.dataclasses import Document
+
+
+@component
+class DocumentLengthRouter:
+    """
+    Categorizes documents based on the length of the `content` field and routes them to the appropriate output.
+
+    A common use case for DocumentLengthRouter is handling documents obtained from PDFs that contain non-text
+    content, such as scanned pages or images. This component can detect empty or low-content documents and route them to
+    components that perform OCR, generate captions, or compute image embeddings.
+
+    ### Usage example
+
+    ```python
+    from haystack_experimental.components.routers import DocumentLengthRouter
+    from haystack.dataclasses import Document
+
+    docs = [
+        Document(content="Short"),
+        Document(content="Long document "*20),
+    ]
+
+    router = DocumentLengthRouter(threshold=10)
+
+    result = router.run(documents=docs)
+    print(result)
+
+    # {
+    #     "short_documents": [Document(content="Short", ...)],
+    #     "long_documents": [Document(content="Long document ...", ...)],
+    # }
+    """
+
+    def __init__(
+        self,
+        *,
+        threshold: int = 10,
+    ) -> None:
+        """
+        Initialize the DocumentLengthRouter component.
+
+        :param threshold:
```

Comment:
Let's specify what unit of length threshold applies to, which appears to be number of characters right?